runtime/queries/column_timeseries.go

```diff
@@ -326,33 +358,18 @@ func (q *ColumnTimeseries) createTimestampRollupReduction(
 		return nil, err
 	}
 	defer rows.Close()
-	results := make([]*runtimev1.TimeSeriesValue, 0, (q.Pixels+1)*4)
-	for rows.Next() {
-		var minT, maxT, argminVT, argmaxVT int64
-		var argminTV, argmaxTV, minV, maxV float64
-		var bin float64
-		err = rows.Scan(&minT, &argminTV, &maxT, &argmaxTV, &minV, &argminVT, &maxV, &argmaxVT, &bin)
-		if err != nil {
-			return nil, err
-		}
-		results = append(results, &runtimev1.TimeSeriesValue{
-			Ts:      time.UnixMilli(minT).Format(IsoFormat),
-			Bin:     &bin,
-			Records: sMap("count", argminTV),
-		}, &runtimev1.TimeSeriesValue{
-			Ts:      time.UnixMilli(argminVT).Format(IsoFormat),
-			Bin:     &bin,
-			Records: sMap("count", minV),
-		}, &runtimev1.TimeSeriesValue{
-			Ts:      time.UnixMilli(argmaxVT).Format(IsoFormat),
-			Bin:     &bin,
-			Records: sMap("count", maxV),
-		}, &runtimev1.TimeSeriesValue{
-			Ts:      time.UnixMilli(maxT).Format(IsoFormat),
-			Bin:     &bin,
-			Records: sMap("count", argmaxTV),
-		})
-		if argminVT > argmaxVT {
+	aggs, err := rowsToData(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]*runtimev1.TimeSeriesValue, 0, len(aggs)*4)
+	for _, v := range aggs {
+		addStruct(v, &results, "min_t", "argmin_tv")
```

Comment:
Feels like using `rowsToData` has added more complexity here. The interplay between `rowsToData` and various re-encoding in `addStruct` makes it hard to figure out exactly what the output will look like (and whether any of the type assertions can error or field names can overlap). The previous `for` loop over `rows` felt more readable. 

Maybe revert to just iterate over `rows` and build the results directly, even it feels a little repetitive?